pmdarima/arima/seasonality.py

```diff
@@ -26,10 +28,124 @@
 
 __all__ = [
     'CHTest',
+    'decompose',
     'OCSBTest'
 ]
 
 
+def decompose(x, type, f, filter=None):
+    """
+
+    Parameters
+    ----------
+    x : np.array
+        The time series of which the trend, seasonal, and noise/random
+        components will be extracted.
+
+    type: str
+        The type of decomposition that will be performed - 'multiplicative' or
+        'additive'. We would use 'multiplicative' generally when we see an
+        increasing trend. We use 'additive' when the trend is relatively
+        stable over time.
+
+    f: int
+        The frequency in terms of number of observations. This behaves
+        similarly to R's frequency for a time series (ts).
+
+    filter: np.array
```

Comment:
Can you make this `filter: np.array, optional (default=None)`? Also, `filter` is another reserved word in Python. Would it make sense to rename to `filt` (or whatever you like)?

Reply:
I will follow PEP8 and just add a single trailing underscore for python reserved keywords. :). See above reply to link to PEP8 docs.

Reply:
Done!